src/game_party.h

```diff
@@ -311,9 +331,56 @@ class Game_Party : public Game_Party_Base {
 	 */
 	bool ApplyStateDamage();
 
+private:
+	const RPG::SaveInventory& data() const;
+	RPG::SaveInventory& data();
+
 private:
 	std::vector<int> state_steps_hp;
 	std::vector<int> state_steps_sp;
 };
 
+// ------ INLINES --------
+
+inline const RPG::SaveInventory& Game_Party::data() const {
+	return Main_Data::game_data.inventory;
+}
+
+inline RPG::SaveInventory& Game_Party::data() {
+	return Main_Data::game_data.inventory;
+}
+
+inline int Game_Party::GetBattleCount() const {
+	return data().battles;
+}
+
+inline void Game_Party::IncBattleCount() {
+	++data().battles;
+}
+
+inline int Game_Party::GetWinCount() const {
+	return data().victories;
+}
+
+inline void Game_Party::IncWinCount() {
+	++data().victories;
+}
+
+inline int Game_Party::GetDefeatCount() const {
+	return data().defeats;
+}
+
+inline void Game_Party::IncDefeatCount() {
+	++data().defeats;
+}
+
+inline int Game_Party::GetRunCount() const {
+	return data().escapes;
+}
+
+inline void Game_Party::IncRunCount() {
+	++data().escapes;
+}
```

Comment:
Educational question:
I wonder about this inline void getter/setter. Are there any C++ references that suggest this? Is there an advantage in putting this in the header instead of cpp file?

Reply:
Inlining is why C++ is so fast and yet enables high level abstractions. Functions which are very small should be inlined for performance. These `Game_Party` methods either just return or increment an integer. There is no benefit to paying the overhead of an entire function call for this.

The real power of this is when things are combined together because it enables the compiler optimizer to do smart things. For example, imagine you had a function `foo(int p)` and the first thing int does is check if `p == 0`. Imagine your calling code sets `int p = 3` and then calls `foo(p)`. If `foo()` is inlined, the compiler can see that `p` can never be zero, and therefore it could remove the entire `if (p == 0) { ... }` block from the code.

A good example is `unique_ptr`. All of `unique_ptr` is inlined, so in most cases if you write a piece of code using `unique_ptr` the compiler will cleanup the memory for you but when the optimizer hits you code the `unique_ptr` gets completely removed and it's as if you wrote `new` and `delete` by hand in all the right places.